client/packages/common/src/ui/components/inputs/Select/Select.tsx

```diff
@@ -17,33 +19,69 @@ const defaultRenderOption = (option: Option) => (
 );
 
 export const Select: FC<SelectProps> = React.forwardRef(
-  ({ options, renderOption, sx, InputProps, ...props }, ref) => (
-    <TextField
-      ref={ref}
-      sx={{
-        '& .MuiInput-underline:before': { borderBottomWidth: 0 },
-        '& .MuiInput-input': { color: theme => theme.palette.gray.dark },
-        ...sx,
-      }}
-      select
-      variant="standard"
-      size="small"
-      InputProps={{
-        disableUnderline: true,
-        ...InputProps,
-        sx: {
-          backgroundColor: theme =>
-            props.disabled
-              ? theme.palette.background.toolbar
-              : theme.palette.background.menu,
-          borderRadius: '8px',
-          padding: '4px 8px',
-          ...InputProps?.sx,
-        },
-      }}
-      {...props}
-    >
-      {options.map(renderOption || defaultRenderOption)}
-    </TextField>
-  )
+  (
+    {
+      options,
+      renderOption,
+      sx,
+      InputProps,
+      clearable = false,
+      value,
+      onChange = () => {},
+      ...props
+    },
+    ref
+  ) => {
+    const SelectProps =
+      clearable && value
+        ? {
+            IconComponent: () => (
+              // Element to clear current selection.
+              <CloseIcon
+                onClick={e =>
+                  onChange({ ...e, target: { value: undefined } } as any)
```

Comment:
Seems like a pretty hacky way to pass a fake "onChange" event to the parent. Any ideas how to make this a bit more elegant (or at least get rid of the `any`)?

Reply:
It's clearer but much too verbose
```
const clearValue = (e: React.MouseEvent<SVGSVGElement, MouseEvent>): void =>
      onChange({
        ...e,
        target: { value: undefined },
      } as unknown as React.ChangeEvent<HTMLInputElement>);
```

and 
```
<CloseIcon
                onClick={clearValue}
...
```

Reply:
You can use your own onChange callback:

```typescript
export interface SelectProps extends Omit<StandardTextFieldProps, 'onChange'> {
  options: Option[];
  renderOption?: (option: Option) => React.ReactNode;
  clearable?: boolean;
  onChange: (value?: string) => void;
}
```

Then you can call it with undefined from the clear and with a string when it comes from a real select input. (untested)